editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java

```diff
@@ -0,0 +1,357 @@
+        }
+      });
+      geodatabase.loadAsync();
+
+      // set on click behaviour
+      mapView.setOnMouseClicked(event -> {
+        // check that the primary mouse button was clicked
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // create a point where the user clicked
+          Point2D screenPoint = new Point2D(event.getX(), event.getY());
+          // call select or move method to move to the point
+          selectOrMove(screenPoint);
+        }
+      });
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Selects the nearest feature, or move the point or polyline vertex to the given screen point.
+   *
+   * @param screenPoint location the user clicked
+   */
+  private void selectOrMove(Point2D screenPoint) {
+
+    // if a feature hasn't been selected, select the feature
+    if (selectedFeature == null) {
+      identifyFeature(screenPoint);
+    } else {
+      // convert the screen point to a map point
+      Point mapPoint = mapView.screenToLocation(screenPoint);
+      // if the feature is a polyline, move the polyline
+      if (selectedFeatureIsPolyline) {
+        movePolylineVertex(mapPoint);
+      } else {
+        // if the feature is a point, move the point
+        movePoint(mapPoint);
+      }
+    }
+  }
+
+  /**
+   * Identifies a feature near the given screen point.
+   *
+   * @param screenPoint location the user clicked
+   */
+  private void identifyFeature(Point2D screenPoint) {
+
+    // clear any previously selected features
+    clearSelection();
+
+    // identify across all layers
+    ListenableFuture<List<IdentifyLayerResult>> identifyLayerResultsFuture = mapView.identifyLayersAsync(screenPoint, 1, false);
+    identifyLayerResultsFuture.addDoneListener(() -> {
+      try {
+        // get the list of results from the future
+        List<IdentifyLayerResult> identifyLayerResults = identifyLayerResultsFuture.get();
+        // if one or more results have been identified
+        if (!identifyLayerResults.isEmpty()) {
+          // retrieve the first result
+          IdentifyLayerResult layerResult = identifyLayerResults.get(0);
+          // check that the result is a feature layer, thereby excluding annotation layers
+          if (layerResult.getLayerContent() instanceof FeatureLayer) {
+            // get a reference to the identified feature
+            selectedFeature = (Feature) layerResult.getElements().get(0);
+            // check the geometry and select the feature
+            selectFeature(layerResult);
+          }
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    });
+  }
+
+  /**
+   * Checks if the identified feature is a straight polyline or a point, and select the feature.
+   * For a point feature, show a dialog to edit attributes. Future clicks will call move functions.
+   *
+   * @param layerResult identify layer result
```

Comment:
Just another suggestion:
```suggestion
   * @param layerResult the identify layer result from which to select a feature
```